src/main/java/org/spongepowered/api/item/FireworkEffectBuilder.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item;
+
+import java.awt.Color;
+
+/**
+ * A builder for {@link FireworkEffect}s.
+ */
+public interface FireworkEffectBuilder {
+
+    /**
+     * Sets whether the {@link FireworkEffect} is going to have a trail
+     * or not.
+     *
+     * @param trail Whether the firework will have a trail
+     * @return This builder, for chaining
+     */
+    FireworkEffectBuilder trail(boolean trail);
+
+    /**
+     * Sets whether the {@link FireworkEffect} is going to flicker
+     * on explosion.
+     *
+     * @param flicker Whether the explosion will flicker
+     * @return This builder, for chaining
+     */
+    FireworkEffectBuilder flicker(boolean flicker);
+
+    /**
+     * Adds the given {@link Color} to the initial explosion colors.
+     *
+     * <p>Colors can be mixed and matched in the order they are added
+     * in.</p>
+     *
+     * @param color The color to add to the explosion
+     * @return This builder, for chaining
+     */
+    FireworkEffectBuilder color(Color color);
+
+    /**
+     * Adds the given {@link Color} to the initial explosion colors.
```

Comment:
Color should be plural in this case since it's an array " Adds the given {@link Color}s ..."
Same for `Iterable<Color>` below